astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,254 @@
+        comment : str
+            comment associated with specified keyword.
+        """
+        if keyword in self.TIME_KEYWORDS:
+            self.global_info[self.TIME_KEYWORDS[keyword]] = value
+        else:
+            raise ValueError('Illegal global time keyword: {} = {}'.format(keyword, value))
+
+    def set_column_override(self, keyword, value, comment=None):
+        """
+        Set the time column specific override attributes.
+
+        Parameters
+        ----------
+        key : str
+            FITS time column specific keyword.
+        value : int, str, list
+            value associated with specified keyword.
+        comment : str
+            comment associated with specified keyword.
+        """
+        if self.is_time_column_keyword(keyword):
+            key, idx = re.match(r'([A-Z]+)([0-9]+)', keyword).groups()
+            self.time_columns[int(idx)][self.COLUMN_TIME_KEYWORDS[key]] = value
+        else:
+            raise ValueError('Illegal column-specific time keyword: {} = {}'.format(keyword, value))
+
+    def convert_time_columns(self, table):
+        """
+        Convert time columns to Astropy Time columns.
+
+        Parameters
+        ----------
+        table : astropy.table.Table
+            The table whose time columns are to be converted.
+        """
+        for idx, time_col in self.time_columns.items():
+            time_colname = table.colnames[idx - 1]
+            table[time_colname] = Time(table[time_colname][...,0], table[time_colname][...,1],
+                                       format='jd', scale=time_col['scale'].lower())
+            # Add location if attribute `pos` is set
+            try:
+                if time_col['pos'] == 'TOPOCENTER':
+                    table[time_colname].location = EarthLocation(self.global_info['loc_x'],
+                                                                 self.global_info['loc_y'],
+                                                                 self.global_info['loc_z'], 
+                                                                 unit='m')
+            except:
+                pass
+
+    def read_time(self, hdr, table):
+        """
+        Read time coordinates in a FITS Binary Table as Time.
+
+        Parameters
+        ----------
+        hdr : `~astropy.io.fits.header.Header`
+            FITS Header
+        table : astropy.table.Table
+            The table whose time columns are to be read as Time
+        Returns
+        -------
+        hdr : `~astropy.io.fits.header.Header`
+            Modified FITS Header (time metadata removed)
+        """
+        # Make a "copy" (not just a view) of the input header, since it
+        # may get modified.  the data is still a "view" (for now)
+        hcopy = hdr.copy(strip=True)
+
+        for key, value, comment in hdr.cards:
+            if (key.upper() in self.TIME_KEYWORDS):
+
+                self.set_global_time(key, value, comment)
```

Comment:
`set_global_time` will also check that `keyword` is in `self.TIME_KEYWORDS`, so this can be simplified (is `set_global_time` still needed ?).